src/single_random_draw.rs

```diff
@@ -68,22 +43,22 @@ pub fn select_coins_srd<R: rand::Rng + ?Sized>(
     let mut value = Amount::ZERO;
 
     for w_utxo in weighted_utxos {
-        let effective_value: Option<Amount> = get_effective_value(w_utxo, fee_rate)?;
+        let utxo_value = w_utxo.utxo.value;
+        let effective_value: SignedAmount = effective_value(fee_rate, w_utxo.satisfaction_weight, utxo_value)?;
 
-        // skip if effective_value is negative.
-        match effective_value {
-            Some(e) => value += e,
-            None => continue,
-        }
+        value += match effective_value.to_unsigned() {
+            Ok(amt) => amt,
+            Err(_) => Amount::ZERO,
```

Comment:
Shouldn't you discard the input if the value is negative instead of counting it as zero?

Reply:
Unless I'm missing something, since you're pushing the utxo to the `result` vector anyway it seems like the total value could be lower than what you compute if you add utxos with negative values.

Reply:
:facepalm: yes, you're right, this is getting pushed to `result` later on which it shouldn't.  Thanks, I'll update this.

Reply:
Ok this has been fixed.  I also expanded the test `select_coins_skip_negative_effective_value`.  Previously, the test had only one utxo which was negative, so it passed.  The test now contains multiple positive effective values and one negative effective value which would then fail if the negative effective value is not discarded.